massa-execution-worker/src/active_history.rs

```diff
@@ -89,4 +90,16 @@ impl ActiveHistory {
         }
         HistorySearchResult::NoInfo
     }
+
+    /// Query every change of the given address datastore at the end of the active history.
+    pub fn fetch_datastore_changes_of(
+        &self,
+        addr: &Address,
+    ) -> Option<SetUpdateOrDelete<LedgerEntry, LedgerEntryUpdate>> {
+        let mut changes = LedgerChanges::default();
+        for output in self.0.iter() {
+            changes.apply(output.state_changes.ledger_changes.clone());
```

Comment:
this is extremely heavy I think

it is accumulating all changes on the ledger for all addresses, plus cloning at every iteration


I think you should restrict it:

```
for output in self.0.iter() {
    changes.apply(output.state_changes.ledger_changes.get_subset(&vec![address]));
    // where get_subset returns a LedgerChanges copy restricted to a set of addresses (if present)
}



```